aexpect/session_ops.py

```diff
@@ -0,0 +1,598 @@
+    <path>.
+
+    :param session: session to run the command on
+    :type session: ShellSession
+    :param str expr: search expression
+    :param str path: file to search
+    :returns: standard output of grep
+    :rtype: str
+    """
+    return grep(session, expr, path, flags=["v"])
+
+
+def grep_vr(session, expr, path):
+    """
+    Invoke ``grep(1)`` on guest searching for the complement of *expr*
+    recursively in *path*.
+
+    :param session: session to run the command on
+    :type session: ShellSession
+    :param str expr: search expression
+    :param str path: file to search
+    :returns: standard output of grep
+    :rtype: str
+    """
+    return grep(session, expr, path, flags=["v", "r"])
+
+
+def grep_q(session, expr, path, flags=None):
+    """
+    Invoke ``grep(1)`` on guest in silent mode searching for *expr* in *path*.
+
+    :param session: session to run the command on
+    :type session: ShellSession
+    :param str expr: search expression
+    :param str path: file to search
+    :param str flags: extra flags passed to ``grep`` on the command line
+    :returns: whether there is a match or not
+    :rtype: bool
+    """
+    if flags is not None:
+        extra = ["-" + flag for flag in flags if flag in GREP_FLAGS_KNOWN]
+        flagstr = " ".join(extra)
+        if extra:
+            return session.cmd("grep -a -q %s '%s' '%s' && echo yes || "
+                               "echo notreally" % (flagstr, expr, path))
+
+    return session.cmd("grep -a -q '%s' '%s' && echo yes || echo notreally"
+                       % (expr, path)).strip() == "yes"
+
+
+def grep_qv(session, expr, path):
+    """
+    Invoke ``grep(1)`` on guest in silent mode searching for the complement of
+    *expr* in *path*.
+
+    :param session: session to run the command on
+    :type session: ShellSession
+    :param str expr: search expression
+    :param str path: file to search
+    :returns: whether there is no match
+    :rtype: bool
+    """
+    return grep_q(session, expr, path, flags=["v"])
+
+
+def grep_qr(session, expr, path):
+    """
+    Invoke ``grep(1)`` on guest in silent mode searching for the complement of
+    *expr* recursively in *path*.
+
+    :param session: session to run the command on
+    :type session: ShellSession
+    :param str expr: search expression
+    :param str path: file to search
+    :returns: whether there is a match or not
+    :rtype: bool
+    """
+    return grep_q(session, expr, path, flags=["r"])
+
+
+def grep_qvr(session, expr, path):
+    """
+    Invoke ``grep(1)`` on guest in silent mode searching for the complement of
+    *expr* recursively in *path*.
+
+    :param session: session to run the command on
+    :type session: ShellSession
+    :param str expr: search expression
+    :param str path: file to search
+    :returns: whether there is no match
+    :rtype: bool
+    """
+    return grep_q(session, expr, path, flags=["v", "r"])
```

Comment:
The amount of possible combinations using grep is huge. In order to keep following this approach, we could have a lot of methods here. I'm not sure if this is bringing some advantages to users.

This probably will increasy the complexity of this module.

Since we have `flags` on the main function, I would prefer to have only one very well documented function.

Reply:
Yes, this one is corrected as I had the same criticism and available in the most recent diff. I will be glad if you could quickly scan through the current diff and possible resolve issues that you saw resolved there like this one.

Reply:
> Since we have flags on the main function, I would prefer to have only one very well documented function.

Couldn't agree more.